test/unit/BPool.t.sol

```diff
@@ -847,37 +847,192 @@ contract BPool_Unit_Bind is BasePoolTest {
 }
 
 contract BPool_Unit_Rebind is BasePoolTest {
-  function test_Revert_NotController() private view {}
+  using LibString for *;
 
-  function test_Revert_NotBound() private view {}
+  struct Rebind_FuzzScenario {
+    address token;
+    uint256 balance;
+    uint256 previousBalance;
+    uint256 denorm;
+    uint256 previousDenorm;
+    uint256 totalWeight;
+  }
 
-  function test_Revert_Finalized() private view {}
+  function _setValues(Rebind_FuzzScenario memory _fuzz) internal {
+    // Create mocks
+    _mockTransferFrom(_fuzz.token);
+    _mockTransfer(_fuzz.token);
 
-  function test_Revert_MinWeight() private view {}
+    // Set token
+    _setRecord(
+      _fuzz.token, BPool.Record({bound: true, index: 0, denorm: _fuzz.previousDenorm, balance: _fuzz.previousBalance})
+    );
+
+    // Set finalize
+    _setFinalize(false);
+    // Set totalWeight
+    _setTotalWeight(_fuzz.totalWeight);
+  }
+
+  function _assumeHappyPath(Rebind_FuzzScenario memory _fuzz) internal pure {
+    vm.assume(_fuzz.token != VM_ADDRESS);
+    vm.assume(_fuzz.token != 0x000000000000000000636F6e736F6c652e6c6f67);
+    vm.assume(_fuzz.balance >= MIN_BALANCE);
+    vm.assume(_fuzz.previousBalance >= MIN_BALANCE);
+    vm.assume(_fuzz.totalWeight >= MIN_WEIGHT);
+    vm.assume(_fuzz.totalWeight <= MAX_TOTAL_WEIGHT - MIN_WEIGHT);
+    _fuzz.previousDenorm = bound(_fuzz.previousDenorm, MIN_WEIGHT, _fuzz.totalWeight);
+    _fuzz.denorm = bound(_fuzz.denorm, MIN_WEIGHT, MAX_TOTAL_WEIGHT - _fuzz.totalWeight);
+  }
 
-  function test_Revert_MaxWeight() private view {}
+  modifier happyPath(Rebind_FuzzScenario memory _fuzz) {
+    _assumeHappyPath(_fuzz);
+    _setValues(_fuzz);
+    _;
+  }
 
-  function test_Revert_MinBalance() private view {}
+  function test_HappyPath(Rebind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    bPool.rebind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
```

Comment:
I'd delete this test case as there are others that already use the `happyPath`, verifying that it works.